iOS/Flows/Conversation/ConversationListCoordinator.swift

```diff
@@ -43,17 +43,19 @@ class ConversationListCoordinator: PresentableCoordinator<Void>, ActiveConversat
 
     override func start() {
         super.start()
-
+        
         self.conversationListVC.onSelectedMessage = { [unowned self] (channelId, messageId, replyId) in
-            self.presentThread(for: channelId,
-                                  messageId: messageId,
-                                  startingReplyId: replyId)
+            if let replyId = replyId {
+                self.presentThread(for: channelId, messageId: messageId, startingReplyId: replyId)
+            } else {
+                self.presentMessageDetail(for: channelId, messageId: messageId)
+            }
```

Comment:
If the selected message is a reply go straight to the thread experience. Otherwise open the message details.